Error occurs when building after adding "hi" or "th" to languages

`docusaurus build` gives

TypeError: Cannot read properties of undefined (reading 'init')
[ERROR] Unable to build website for locale en.
[ERROR] Error: Failed to compile with errors.
    at C:\dev\bloom-docs\node_modules\@docusaurus\core\lib\webpack\utils.js:180:24
    at C:\dev\bloom-docs\node_modules\webpack\lib\MultiCompiler.js:631:14
    at processQueueWorker (C:\dev\bloom-docs\node_modules\webpack\lib\MultiCompiler.js:568:6)
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
[INFO] Docusaurus version: 2.4.3
Node version: v18.16.0
error Command failed with exit code 1.

Seems to be caused by https://github.com/MihaiValentin/lunr-languages/issues/101.

Note that while I am using docusaurus v2, this error occurs with v3 as well.
I used https://docusaurus.new/, just added the dependency, and added 
```
  themes: [
    [
      require.resolve("@easyops-cn/docusaurus-search-local"),
      {
        language: ["en", "hi"]
      },
    ],
  ],
```
to the config. 

That gave:  
```
[ERROR] Error: Unable to build website for locale en.
    at tryToBuildLocale (/project/workspace/node_modules/@docusaurus/core/lib/commands/build.js:54:19)
    at async /project/workspace/node_modules/@docusaurus/core/lib/commands/build.js:65:9
    at async mapAsyncSequential (/project/workspace/node_modules/@docusaurus/utils/lib/jsUtils.js:20:24)
    at async Command.build (/project/workspace/node_modules/@docusaurus/core/lib/commands/build.js:63:5) {
  [cause]: TypeError: Cannot read properties of undefined (reading 'init')
      at server.bundle.js:16693:15
      at 1088 (server.bundle.js:14194:70)
      at __webpack_require__ (server.bundle.js:28127:42)
      at 7401 (server.bundle.js:11585:24)
      at __webpack_require__ (server.bundle.js:28127:42)
      at 7489 (server.bundle.js:8814:14)
      at __webpack_require__ (server.bundle.js:28127:42)
      at 6533 (server.bundle.js:9695:21)
      at __webpack_require__ (server.bundle.js:28127:42)
      at server.bundle.js:28193:37
}
```

--- a/docusaurus-search-local/src/server/utils/buildIndex.ts
+++ b/docusaurus-search-local/src/server/utils/buildIndex.ts
@@ -6,6 +6,7 @@ import {
   SearchDocument,
   WrappedIndex,
 } from "../../shared/interfaces";
+import { LANGUAGES_NEED_WORDCUT } from "../../shared/constants";
 
 let pluginInitialized = false;
 let plugin: lunr.Builder.Plugin | undefined;
@@ -27,6 +28,9 @@ export function buildIndex(
     }
     if (language.includes("ja") || language.includes("jp")) {
       require("lunr-languages/tinyseg")(lunr);
+    }
+    if (LANGUAGES_NEED_WORDCUT.some((item) => language.includes(item))) {
+      (lunr as any).wordcut = require("lunr-languages/wordcut");
     }
     for (const lang of language.filter(
       (item) => item !== "en" && item !== "zh"

--- a/docusaurus-search-local/src/server/utils/generate.ts
+++ b/docusaurus-search-local/src/server/utils/generate.ts
@@ -2,6 +2,7 @@ import fs from "fs";
 import path from "path";
 import { ProcessedPluginOptions } from "../../shared/interfaces";
 import { getIndexHash } from "./getIndexHash";
+import { LANGUAGES_NEED_WORDCUT } from "../../shared/constants";
 
 export function generate(config: ProcessedPluginOptions, dir: string): string {
   const {
@@ -120,6 +121,13 @@ export function generate(config: ProcessedPluginOptions, dir: string): string {
       `require(${JSON.stringify(
         require.resolve("lunr-languages/tinyseg")
       )})(lunr);`
+    );
+  }
+  if (language.some((item) => LANGUAGES_NEED_WORDCUT.includes(item))) {
+    constantContents.push(
+      `lunr.wordcut = require(${JSON.stringify(
+        require.resolve("lunr-languages/wordcut")
+      )});`
     );
   }
   for (const lang of language.filter(

--- a/docusaurus-search-local/src/shared/constants.ts
+++ b/docusaurus-search-local/src/shared/constants.ts
@@ -1,0 +1,1 @@
+export const LANGUAGES_NEED_WORDCUT = ["th", "hi", "te", "ta", "kn", "sa"];

